### PR TITLE
feat: pages state, buttons

### PR DESCRIPTION
**Pull request by botjoker**
- Implement page state management to preserve scroll positions and other stateful data across page navigations.
- Add a new Button component for reusable UI interactions.
- Introduce a Link component to enhance navigation with styled links.
- Apply minor adjustments on the Agents and Abilities pages for improved user experience.

## Summary by CodeRabbit

- **New Features**
    - Introduced functionality to persist and restore scroll positions across pages.
    - Added search functionality with a search input field in various sections.
    - Enhanced `BaseButton` component with new properties like `color`, `shade`, and `size`.
    - Introduced a customizable `Link` component with properties for color, size, and disabled state.
- **Refactor**
    - Adjusted styles for search inputs, entity switches, and main content headers.
    - Modified `BaseButton` usage in forms and item views with new color and size properties.
- **Style**
    - Updated border thickness and color adjustments for the `ToggleSwitch` component.
- **Documentation**
    - Expanded module functionality with state management capabilities and a simple hashing algorithm.

### CODE REVIEW

File: shared/lib/pageState.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  pageStates: Record<string, PageState>
+}
+
+export const usePageStateStore = defineStore('pageState', {
+  state: (): State => ({
+    pageStates: {},
+  }),
+  actions: {
+    saveState(routePath: string, state: PageState) {
+      const hash = simpleHash(routePath)
```
[the remark this answers is not in the document]
Comment (rkorzhoff) [reply shared/lib/pageState.ts:21]: Also, In our scenario, a Queue data structure would be more suitable, as it enables us to cap the maximum amount of data we retain. Let's set a limit of 50 routes.

File: shared/lib/useSavePageState.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2024 StarfleetAI
+// SPDX-License-Identifier: Apache-2.0
+
+import { onMounted, onUnmounted, ref } from 'vue'
+import { useRoute } from 'vue-router'
+import { usePageStateStore, type PageState } from '@/shared/lib/pageState'
```
Comment (rkorzhoff) [inline shared/lib/useSavePageState.ts:6]: Imports shouldn't target directly to files when use path aliases, instead use relative path `./pageState`. Check other places too pls

File: shared/lib/useSavePageState.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+export const useSavePageState = () => {
+  const pageStateStore = usePageStateStore()
+  const route = useRoute()
+  const savedState = ref<PageState>({ scrollPosition: 0 })
+
+  onMounted(() => {
+    const state = pageStateStore.getState(route.path)
+    if (state) {
+      savedState.value = state
+      window.scrollTo(0, state.scrollPosition)
```
Comment (rkorzhoff) [inline shared/lib/useSavePageState.ts:17]: We don't use the window scroll; instead, we implement scrolling for individual blocks within the page

File: shared/ui/base/BaseButton.vue [hunk 4]
```diff
@@ -4,77 +4,171 @@
...
       disabled: false,
-      outlined: false,
       size: 'medium',
     },
   )
 </script>
 
 <template>
-  <div :class="['base-button', type, size, { disabled, outlined }]">
+  <div :class="['base-button', color, shade, size, type, { disabled }]">
```
Comment (rkorzhoff) [inline shared/ui/base/BaseButton.vue:24]: Replace with `<button>` tag to achieve better navigation

File: shared/ui/base/Link.vue [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      color: 'blue',
+      shade: 'default',
+      disabled: false,
+      size: 'medium',
+    },
+  )
+</script>
+
+<template>
+  <div :class="['link', color, size, { disabled }]">
```
Comment (rkorzhoff) [inline shared/ui/base/Link.vue:22]: Same here

**Review by rkorzhoff — CHANGES_REQUESTED** [review]
Everything else is ok!

File: shared/lib/useSavePageState.ts (hunk 2 above)
Comment (botjoker) [reply shared/lib/useSavePageState.ts]: I fixed it

File: shared/lib/useSavePageState.ts (hunk 3 above)
Comment (botjoker) [reply shared/lib/useSavePageState.ts:17]: This is just for example. Real scroll I'll add later.

File: shared/ui/base/BaseButton.vue (hunk 4 above)
Comment (botjoker) [reply shared/ui/base/BaseButton.vue]: I fixed it

File: shared/lib/pageState.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (botjoker) [reply shared/lib/pageState.ts]: I fixed it

**Review by rkorzhoff — APPROVED** [review]

**Pull request merged**